### DISCUSSION TITLE
Is there any way to start or stop HSL recording on the fly for specific stream Id?

### DISCUSSION THREAD

**Discussion by enkhchuluun** [ant-media/Ant-Media-Server#5479] [FAQs] [answered]
I have a requirement as follow:
- our system provide a stream URL and stream key (stream Id) for our publishers
- a publisher start push their stream using OBS like tool
- the publisher can see and verify his stream on our website (using antmedia WebRTC client player)
- then the publisher click start button to publish their stream to public
- when the public stream is started then the HSL recording should be started

Can you help how to implemet this kind of feature?
Thanks

**Comment by Mohit-3196** [comment, marked as answer]
Hi @enkhchuluun ,
We have start and stop REST API for MP4 and WebM recording https://antmedia.io/rest/#/BroadcastRestService/enableRecording

For the HLS recordings, for now, they can be configured to record the streams as soon as the stream starts.
If this doesn't suit your requirements then I welcome you to create a feature request here https://github.com/ant-media/Ant-Media-Server/issues and we can discuss it with you in more detail and implement it.

Thank you.

**Reply by yashtandon113** [reply]
@Mohit-3196 @enkhchuluun 
We have a feature request for the HLS recording control and it is under backlog. We can discuss internally to get it scheduled according to the priority.
https://github.com/ant-media/Ant-Media-Server/issues/5425
